src/bp-core/bp-core-rewrites.php

```diff
@@ -187,6 +187,8 @@ function bp_rewrites_get_url( $args = array() ) {
 		)
 	);
 
+	$r = $args;
```

Comment:
Here we should have:
```suggestion
	// Define the path chunks out of parsed arguments to make them available & unchanged for the 'bp_rewrites_get_url' filter.
	$path_chunks = $r;
```